src/sonic-config-engine/sonic_device_util.py

```diff
@@ -125,6 +133,78 @@ def get_all_namespaces():
 
     return {'front_ns':front_ns, 'back_ns':back_ns}
 
+def get_port_config_from_all_asics(): 
+    if not is_multi_npu():
+        return get_port_config()
+
+    platform = get_platform()
+    hwsku = get_hwsku()
+    all_ports = {}
+    ns_list  = get_namespaces()
+    
+    for ns in ns_list:
+        asic_id = get_npu_id_from_name(ns)
+        (ports, _, _) = get_port_config(hwsku=hwsku, platform=platform, asic=asic_id)
+        all_ports.update(ports)
+    
+    return all_ports
+    
+def get_port_role(port_name):
+    ports_config = get_port_config_from_all_asics()
+    if not ports_config[port_name].has_key('role'):
+        return EXTERNAL_PORT
+    
+    role = ports_config[port_name]['role']
+    return role
+
+def is_port_internal(port_name):
+    role = get_port_role(port_name)
+    if role is INTERNAL_PORT:
+        return True
+    return False
+
+def is_port_channel_internal(port_channel, namespace=None):
+    if not is_multi_npu():
+        return False
+    if namespace is None:
+        ns_list  = get_namespaces()
+    else:
+        ns_list = [namespace]
+        
+    for ns in ns_list:
+        config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=ns)
+        config_db.connect()
+        #import pdb; pdb.set_trace()
+        port_channels = config_db.get_table(PORT_CHANNEL_CFG_DB_TABLE)
+        if not port_channels.has_key(port_channel):
+            continue
+        members = port_channels[port_channel]['members']
+        for member in members:
+            if is_port_internal(member):
+                return True
```

Comment:
This logic will be intensive for 'external' ports. It will 'get_port_config_from_all_asics()' for all ports. Any reason why you are checking all member of 'external' role and only one member for 'internal' role. 

will this work:

return is_port_internal(members[0])

Reply:
Fixed in the latest commit